pkg/xray/resource_xray_repository_config.go

```diff
@@ -0,0 +1,349 @@
+			RepoName: d.GetString("repo_name", false),
+		}
+
+		if _, ok := s.GetOk("repo_config"); ok {
+			repoConfig := unpackRepoConfig(s.Get("repo_config").(*schema.Set))
+			repositoryConfig.RepoConfig = repoConfig
+		}
+
+		if _, ok := s.GetOk("repo_paths_config"); ok {
+			repoPathConfig := unpackRepoPathConfig(s.Get("repo_paths_config").(*schema.Set))
+			repositoryConfig.RepoPathsConfig = repoPathConfig
+		}
+		return repositoryConfig
+	}
+
+	var packGeneralRepoConfig = func(repoConfig *GeneralRepoConfig) []interface{} {
+		if repoConfig == nil {
+			return []interface{}{}
+		}
+
+		m := map[string]interface{}{}
+		m["vuln_contextual_analysis"] = repoConfig.VulnContextualAnalysis
+		m["retention_in_days"] = repoConfig.RetentionInDays
+
+		return []interface{}{m}
+	}
+
+	var packAllOtherArtifacts = func(otherArtifacts AllOtherArtifacts) []interface{} {
+		m := map[string]interface{}{
+			"index_new_artifacts": otherArtifacts.IndexNewArtifacts,
+			"retention_in_days":   otherArtifacts.RetentionInDays,
+		}
+
+		return []interface{}{m}
+	}
+
+	var packPatterns = func(patterns []Pattern) []interface{} {
+		var ps []interface{}
+
+		for _, pattern := range patterns {
+			p := map[string]interface{}{
+				"include":             pattern.Include,
+				"exclude":             pattern.Exclude,
+				"index_new_artifacts": pattern.IndexNewArtifacts,
+				"retention_in_days":   pattern.RetentionInDays,
+			}
+
+			ps = append(ps, p)
+		}
+
+		return ps
+	}
+
+	var packRepoPathsConfigList = func(repoPathsConfig *RepoPathsConfiguration) []interface{} {
+		if repoPathsConfig == nil {
+			return []interface{}{}
+		}
+
+		m := map[string]interface{}{}
+		m["pattern"] = packPatterns(repoPathsConfig.PatternList)
+		m["all_other_artifacts"] = packAllOtherArtifacts(repoPathsConfig.OtherArtifacts)
+
+		return []interface{}{m}
+	}
+
+	var packRepositoryConfig = func(ctx context.Context, repositoryConfig RepositoryConfiguration, d *schema.ResourceData) diag.Diagnostics {
+		if err := d.Set("repo_name", repositoryConfig.RepoName); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("repo_config", packGeneralRepoConfig(repositoryConfig.RepoConfig)); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("repo_paths_config", packRepoPathsConfigList(repositoryConfig.RepoPathsConfig)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		return nil
+	}
+
+	var resourceXrayRepositoryConfigRead = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		repositoryConfig := RepositoryConfiguration{}
+
+		resp, err := m.(*resty.Client).R().SetResult(&repositoryConfig).
+			SetPathParams(map[string]string{
+				"repo_name": d.Id(),
+			}).
+			Get("xray/api/v1/repos_config/{repo_name}")
+
+		if err != nil {
+			if resp != nil && resp.StatusCode() != http.StatusOK {
+				tflog.Error(ctx, fmt.Sprintf("Request payload is invalid as repo (%s) is either not indexed or does not exist", d.Id()))
```

Comment:
This is a GET so there's no request payload. Why log the error only for this condition?

Reply:
This check for the case if the repo doesn't exist and the error message is a copy of the API response (almost). I just added the repo name.
Changed to `Repo (%s) is either not indexed or does not exist`, its more clear.